src/hypothesis/internal/compat.py

```diff
@@ -502,10 +502,16 @@ def implements_iterator(it):
 
 if PY2:
     def floor(x):
-        return int(math.floor(x))
+        # `int(math.floor(x))` fails for negative long integers
```

Comment:
FWIW I think encoding some of this in the comment would be a good idea. Comments of the form "fails" without saying what fails are basically an invitation to future debugging fun and games.